src/Struct/Exprs.hs

```diff
@@ -19,9 +19,12 @@ data UsProg =
   | UsProgData Var [Var] [Ctor]         -- lhs, type params, constructors
   deriving (Eq, Ord)
 
+data Forall = Forall Var Bool -- name, is robust
+  deriving (Eq, Ord)
+
 -- Scheme-ified definition
 data SProg =
-    SProgDefine Var [Var] [Var] Term Type  -- lhs, tags, type params, rhs, type
+    SProgDefine Var [Var] [Forall] Term Type  -- lhs, tags, type params, rhs, type
```

Comment:
Is it undesirable that technically the tags are all also \forall-quantified but we don't call them "Forall"?

Reply:
It's as if what's called `Forall` here is really `ForallType`, and the bracketed `Var` here is really `ForallTag`, but we just don't have any constraints we want to put on a tag variable...yet